packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -163,21 +159,24 @@ export type ProtocolDetail<
  * itself can fail. In that case, publishes final asset location to vstorage
  * and gives up. Clients must manually rebalance to recover.
  */
-export const trackFlow = async (
+const trackFlow = async (
   reporter: GuestInterface<PortfolioKit['reporter']>,
   todo: (() => Promise<AssetMovement>)[],
+  portfolioTrace: TraceLogger,
 ) => {
   const flowId = reporter.allocateFlowId();
+  const flowTrace = portfolioTrace.sub(`flow${flowId}`);
   let step = 1;
   const moves: AssetMovement[] = [];
   try {
     for (const makeMove of todo) {
+      const stepTrace = flowTrace.sub(`step${step}`);
```

Comment:
"trace" is a verb and "step" is the direct object (what it is tracing)

```suggestion
      const traceStep = flowTrace.sub(`step${step}`);
```

I request that we establish a coding style with the new `.sub` feature that we maintain `trace` as the start of the function name and suffix each Sub`. E.g. `trace`, `traceFlow`, `traceFlowStep`. Perhaps the intermediate nouns can be elided when there's more than two subs.

Reply:
using verbs makes sense, especially for functions whose purpose is a side-effect.

For things with smallish scope such as the inside of this loop, I find two words (total) to be plenty, though.

Reply:
I agree with that and retract the suggestion that nouns be nested. I will point out that the `traceFoo` casing makes it more consistent when someone does want to use multiple nouns: `traceFlowStep` > `flowStepTrace`

Reply:
I like these suggestions taken together. Done.